Crashes when publiccode.yml is a symlink or a directory

DownloadURL is nil when publiccode.yml in the repo is a symlink to a directory or a directory.

We don't have such a case in the catalog, but it's reproducible adding https://github.com/italia/publiccode.yml-docs to the list of onboarded software.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xa9a0a3]

goroutine 102 [running]:
github.com/italia/publiccode-crawler/v3/scanner.GitHubScanner.ScanRepo({0xc00026cc00?, {0xe5a2c8?, _}}, {{0xc0014516e0, 0x5}, {0x0, 0x0}, 0x0, {0xc0014516e8, 0xa}, ...}, ...)
	/home/fabio/dev/publiccode-crawler/scanner/github.go:147 +0x583
github.com/italia/publiccode-crawler/v3/scanner.GitHubScanner.ScanGroupOfRepos({0xc00026cc00?, {0xe5a2c8?, _}}, {{0xc000a1a3c0, 0x5}, {0x0, 0x0}, 0x0, {0xc000a1a3c8, 0xa}, ...}, ...)
	/home/fabio/dev/publiccode-crawler/scanner/github.go:76 +0x4fd
github.com/italia/publiccode-crawler/v3/crawler.(*Crawler).ScanPublisher(0xc000228000, {{0xc000b4c090, 0x24}, {0xc0002154f0, 0x42}, {0xc00023ca20, 0x1, 0x1}, {0x0, 0x0, ...}})
	/home/fabio/dev/publiccode-crawler/crawler/crawler.go:173 +0x922
created by github.com/italia/publiccode-crawler/v3/crawler.(*Crawler).CrawlPublishers
	/home/fabio/dev/publiccode-crawler/crawler/crawler.go:122 +0x2a5
```

--- a/scanner/github.go
+++ b/scanner/github.go
@@ -136,6 +136,11 @@ Retry:
 
 		return fmt.Errorf("[%s]: failed to get publiccode.yml: %w", *repo.FullName, err)
 	}
+
+	if file.DownloadURL == nil {
+		return fmt.Errorf("[%s]: failed to get publiccode.yml: not a regular file?", *repo.FullName)
+	}
+
 	if file != nil {
 		canonicalURL, err := url.Parse(*repo.CloneURL)
 		if err != nil {

